newIDE/app/src/ProjectsStorage/LocalProjectWriter.js

```diff
@@ -80,6 +134,38 @@ export default class LocalProjectWriter {
     }
   };
 
+  static saveProjectAs = (project: gdProject): Promise<void> => {
+    const filepath = project.getProjectFile();
+    const filenameWithExtension = path.basename(project.getProjectFile()) || "game.json";
+    const projectPath = path.dirname(project.getProjectFile());
+    const fileSystem = assignIn(new gd.AbstractFileSystemJS(), localFileSystem);
+    const browserWindow = electron.remote.getCurrentWindow();
+    const options = {
+    defaultPath: path.join(projectPath, filenameWithExtension),
+    filters: [{ name: 'GDevelop 5 project', extensions: ['json'] }],
+    };
+
+    if (!dialog) {
+    return Promise.reject('Unsupported');
+    }
+    const newFilepath = dialog.showSaveDialog(browserWindow, options);
+    if (!newFilepath || newFilepath === '') {
+    return Promise.reject('Filepath from dialog is empty');
```

Comment:
Actually this should not be considered as an error (the user can cancel and should not see any error message reported).
So you should call instead `return Promise.resolve();`. This will still display the snack about the successful save, which is weird for the user ("what, i did press cancel, why it has being saved"), so return a boolean indicating if the save was really made or not:
```
    return Promise.resolve(false); // User cancelled the save.
```

*See also my other comments about handling this boolean.